exec/testing/fake_exec.go

```diff
@@ -81,6 +87,9 @@ func (fake *FakeExec) CommandContext(ctx context.Context, cmd string, args ...st
 
 // LookPath is for finding the path of a file
 func (fake *FakeExec) LookPath(file string) (string, error) {
+	fake.mu.Lock()
```

Comment:
Can this lead to a lock ordering inversion or even just a recursive lock?

Reply:
`Command()` and `LookPath()` don't explicitly call each other. This means that once either function acquires a lock, it will be able to finish and unlock. Still, the caller could embed one method inside the other, which would cause deadlock.

I went ahead and tightened the lock to only the section violating the invariant, i.e. when CommandCalls is incremented and CommandScript accessed non-atomically.

There's still the implication that the fields in the fake object shouldn't be modified while it's being used in a concurrent context, but there are easier ways to make a brittle test.

Reply:
Thanks, I would also at least mention this in the documentation for the function that the risk exists.

Reply:
Good idea. Done.